images/linux/scripts/installers/go.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+################################################################################
+##  File:  go.sh
+##  Desc:  Installs go, configures GOROOT, and adds go to the path
+################################################################################
+
+# Source the helpers for use with the script
+source $HELPER_SCRIPTS/document.sh
+golangTags="/tmp/golang_tags.json"
+# This function installs Go using the specified arguments:
+#   $1=MajorVersion (1.11)
+#   $2=IsDefaultVersion (true or false)
+function InstallGo () {
+    version=$( getFullGoVersion $1 )
+    echo "Install Go $version"
+    curl -sL https://dl.google.com/go/$version.linux-amd64.tar.gz -o $version.linux-amd64.tar.gz
+    mkdir -p /usr/local/go$1
+    tar -C /usr/local/go$1 -xzf $version.linux-amd64.tar.gz --strip-components=1 go
```

Comment:
can we create a variables for `/usr/local/go$1` and `$version.linux-amd64.tar.gz`? it is repeated multiple times in this function and creates some mess with all these slashes and dots. Something like this look a bit more better I think
```
    version=$( getFullGoVersion $1 )
    downloadVersion=$version.linux-amd64.tar.gz
    goFolder=/usr/local/go$1

    echo "Install Go $version"
    curl -sL https://dl.google.com/go/$downloadVersion -o $downloadVersion
    mkdir -p $goFolder
    tar -C $goFolder -xzf $downloadVersion --strip-components=1 go
    rm $downloadVersion
    echo "GOROOT_${1//./_}_X64=$goFolder" | tee -a /etc/environment
    DocumentInstalledItem "Go $1 ($($goFolder/bin/go version))"

    # If this version of Go is to be the default version,
    # symlink it into the path and point GOROOT to it.
    if [ $2 = true ]
    then
        ln -s $goFolder/bin/* /usr/bin/
        echo "GOROOT=$goFolder" | tee -a /etc/environment
    fi
```

Reply:
Done.